src/components/util/overlays/popover/usePopover.ts

```diff
@@ -79,6 +79,47 @@ export const usePopover = <E extends HTMLElement>(
     }
   }, [controller.activate, controller.deactivate]);
   
+  // Added for outside click detection
+  React.useEffect(() => {
+    if (!controller.active) {
+      return;
+    }
+
+    const handleClickOutside = (event: MouseEvent) => {
+      const target = event.target;
+
+      if (!(target instanceof HTMLElement)) {
+        return;
+      }
+
+      const popover = popoverRef.current;
+      
+      // Prevent popover from closing if click was inside the popover
+      if (!popover || popover.contains(target)) {
+        return;
+      }
+
+      // Prevent popover from closing if click was inside portal
+      const isInPortal = target.closest('[data-portal]');
+      if (isInPortal) {
+        return;
+      }
+
+      // Otherwise, clicked outside both → close the popover
+      controller.deactivate();
+    };
+    
+    // Listen for outside clicks
+    document.addEventListener('mousedown', handleClickOutside, true);
+    
+    // Clean up when component unmounts or controller becomes inactive
+    return () => {
+      document.removeEventListener('mousedown', handleClickOutside, true);
+    };
+  }, [controller.active, controller.deactivate]);
```

Comment:
The logic here has an issue: it will assume that _any_ content inside a portal should not close the popover. But this should only hold for portals that came from within the popover itself.